common/src/main/java/com/gregtechceu/gtceu/integration/emi/oreprocessing/GTEmiOreProcessing.java

```diff
@@ -0,0 +1,29 @@
+package com.gregtechceu.gtceu.integration.emi.oreprocessing;
+
+import com.gregtechceu.gtceu.GTCEu;
+import com.gregtechceu.gtceu.api.data.chemical.material.Material;
+import com.gregtechceu.gtceu.integration.GTOreProcessingWidget;
+import com.lowdragmc.lowdraglib.emi.ModularEmiRecipe;
+import com.lowdragmc.lowdraglib.gui.widget.WidgetGroup;
+import dev.emi.emi.api.recipe.EmiRecipeCategory;
+import net.minecraft.resources.ResourceLocation;
+import org.jetbrains.annotations.Nullable;
+
+public class GTEmiOreProcessing extends ModularEmiRecipe<WidgetGroup> {
+    final Material material;
+
+    public GTEmiOreProcessing(Material material) {
+        super(() -> new GTOreProcessingWidget(material));
+        this.material = material;
+    }
+
+    @Override
+    public EmiRecipeCategory getCategory() {
+        return GTOreProcessingEmiCategory.CATEGORY;
+    }
+
+    @Override
+    public @Nullable ResourceLocation getId() {
+        return GTCEu.id("ore_processing_diagram");
```

Comment:
this should probably return `GTCEu.id(material.getName())` according to the EMI docs?